oshdb-api/src/main/java/org/heigit/ohsome/oshdb/api/mapreducer/MapReducer.java

```diff
@@ -754,17 +754,44 @@ protected MapReducer(MapReducer<?> obj) {
     // apply geometry filter as first map function
     final List<MapFunction> remainingMappers = List.copyOf(ret.mappers);
     ret.mappers.clear();
-    if (ret.forClass.equals(OSMContribution.class)) {
-      ret = ret.filter(x -> {
-        OSMContribution c = (OSMContribution) x;
-        return f.applyOSMGeometry(c.getEntityBefore(), c::getGeometryBefore)
-          || f.applyOSMGeometry(c.getEntityAfter(), c::getGeometryAfter);
-      });
-    } else if (ret.forClass.equals(OSMEntitySnapshot.class)) {
-      ret = ret.filter(x -> {
-        OSMEntitySnapshot s = (OSMEntitySnapshot) x;
-        return f.applyOSMGeometry(s.getEntity(), s::getGeometry);
-      });
+    if (this.grouping == Grouping.NONE) {
+      // no grouping -> directly filter using the geometries of the snapshot / contribution
+      if (ret.forClass.equals(OSMEntitySnapshot.class)) {
+        ret = ret.filter(x -> {
+          OSMEntitySnapshot s = (OSMEntitySnapshot) x;
+          return f.applyOSMGeometry(s.getEntity(), s::getGeometry);
+        });
+      } else if (ret.forClass.equals(OSMContribution.class)) {
```

Comment:
not covered by any tests

Reply:
huh, true. Strictly speaking this part of the code wasn't really changed in this PR. but I guess it would be good to have some basic test for this path as well.

Reply:
oh gosh, this branch was actually pretty buggy and slipped through because it wasn't covered by the tests. :see_no_evil: thanks for spotting that!

should be fixed in d468552e